plugins/kotlin/jvm-debugger/core/src/org/jetbrains/kotlin/idea/debugger/core/KotlinPositionManager.kt

```diff
@@ -293,11 +293,8 @@ class KotlinPositionManager(private val debugProcess: DebugProcess) : MultiReque
         return result
     }
 
-    private fun Method.getInlineFunctionBorders(sourceFileName: String): List<ClosedRange<Location>> {
-        return getInlineFunctionOrArgumentVariables()
-            .mapNotNull { it.getBorders() }
-            .filter { it.start.safeSourceName() == sourceFileName }
-            .toList()
+    private fun Method.getInlineFunctionBorders(): List<ClosedRange<Location>> {
```

Comment:
# Why source file name check is not important
Consider the example:
```
inline fun foo(f: () -> Unit) {
    f()
}

fun main() {
    foo {
        println()
    }
}
```

The  bytecode:
```
public static final void main();
    descriptor: ()V
    flags: (0x0019) ACC_PUBLIC, ACC_STATIC, ACC_FINAL
    Code:
      stack=1, locals=2, args_size=0
         0: iconst_0
         1: istore_0
         2: iconst_0
         3: istore_1
         4: getstatic     #33                 // Field java/lang/System.out:Ljava/io/PrintStream;
         7: invokevirtual #38                 // Method java/io/PrintStream.println:()V
        10: nop
        11: nop
        12: nop
        13: return
      LineNumberTable:
        line 7: 0
        line 13: 2
        line 8: 4
        line 9: 10
        line 13: 11
        line 14: 12
        line 10: 13
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
            4       7     1 $i$a$-foo-MainKt$main$1   I
            2      11     0 $i$f$foo   I
```

According to LVT, the inlined function starts at opcode number 2, the lambda starts at opcode 4. As we can see from line number table:
```
 line 13: 2
 line 8: 4
```
these opcodes have a dedicated line number inside a method where the function and lambda are inlined. If my logic is correct, It means that the following check: `it.start.safeSourceName() == sourceFileName` doesn't do anything. Also no tests fail if the check is removed.

# Why removing this check is necessary
Consider the following example:
```
data class A(val x: Int, val y: Int)

inline fun foo(f: (A) -> Unit) {
    f(A(0, 0))
}

fun main() {
    foo { (x, y) ->
        println()
    }
}
```

The bytecode:
```
public static final void main();
    descriptor: ()V
    flags: (0x0019) ACC_PUBLIC, ACC_STATIC, ACC_FINAL
    Code:
      stack=4, locals=5, args_size=0
         0: iconst_0
         1: istore_0
         2: new           #18                 // class A
         5: dup
         6: iconst_0
         7: iconst_0
         8: invokespecial #22                 // Method A."<init>":(II)V
        11: astore_1
        12: iconst_0
        13: istore_2
        14: aload_1
        15: invokevirtual #37                 // Method A.component1:()I
        18: istore_3
        19: aload_1
        20: invokevirtual #40                 // Method A.component2:()I
        23: istore        4
        25: getstatic     #46                 // Field java/lang/System.out:Ljava/io/PrintStream;
        28: invokevirtual #51                 // Method java/io/PrintStream.println:()V
        31: nop
        32: nop
        33: nop
        34: return
      LineNumberTable:
        line 9: 0
        line 15: 2
        line 9: 14
        line 10: 25
        line 11: 31
        line 15: 32
        line 16: 33
        line 12: 34
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
           14      18     2 $i$a$-foo-MainKt$main$1   I
           19      13     3     x   I
           25       7     4     y   I
            2      32     0 $i$f$foo   I
```

From the LVT it is clear that the inlined lambda starts at opcode number 14. From the line number table we can see that this opcode has a dedicated line number:
```
line 9: 14
```

According to the new debug information format for lambdas with parameter destructuring, this line number will be erased by the compiler. Which means that the start of the marker variable `$i$a$-foo-MainKt$main$1` will have a source file of a previous line number `line 15: 2`.  For example it could be `fake.kt` or  something else. In the end it means that `Method.getInlineFunctionBorders(sourceFileName: String)` just wouldn't return this border, as its start would have a different source file, which eventually would lead to a breakpoint being skipped.